nautobot/extras/signals.py

```diff
@@ -55,55 +61,50 @@ def _handle_changed_object(sender, instance, raw=False, **kwargs):
     """
     Fires when an object is created or updated.
     """
-    from .jobs import enqueue_job_hooks  # avoid circular import
 
     if raw:
         return
 
     if change_context_state.get() is None:
         return
 
-    object_m2m_changed = False
-
     # Determine the type of change being made
     if kwargs.get("created"):
         action = ObjectChangeActionChoices.ACTION_CREATE
     elif "created" in kwargs:
         action = ObjectChangeActionChoices.ACTION_UPDATE
     elif kwargs.get("action") in ["post_add", "post_remove"] and kwargs["pk_set"]:
         # m2m_changed with objects added or removed
-        object_m2m_changed = True
         action = ObjectChangeActionChoices.ACTION_UPDATE
     else:
         return
 
     # Record an ObjectChange if applicable
     if hasattr(instance, "to_objectchange"):
+        user = _get_user_if_authenticated(change_context_state.get().get_user(), instance)
         # save a copy of this instance's field cache so it can be restored after serialization
         # to prevent unexpected behavior when chaining multiple signal handlers
         original_cache = instance._state.fields_cache.copy()
-        if object_m2m_changed:
-            related_changes = ObjectChange.objects.filter(
-                changed_object_type=ContentType.objects.get_for_model(instance),
-                changed_object_id=instance.pk,
-                request_id=change_context_state.get().change_id,
-            )
-            m2m_changes = instance.to_objectchange(action)
-            if related_changes.exists():
-                related_changes.update(object_data=m2m_changes.object_data, object_data_v2=m2m_changes.object_data_v2)
-                objectchange = related_changes.first()
-            else:
-                objectchange = m2m_changes
-                objectchange.user = _get_user_if_authenticated(change_context_state.get().get_user(), objectchange)
-                objectchange.request_id = change_context_state.get().change_id
-                objectchange.change_context = change_context_state.get().context
-                objectchange.change_context_detail = change_context_state.get().context_detail[
-                    :CHANGELOG_MAX_CHANGE_CONTEXT_DETAIL
-                ]
-                objectchange.save()
+
+        # if a change already exists for this change_id, user, and object, update it instead of creating a new one
+        # if the object was deleted and then recreated with the same pk (don't do this), change the action to update
```

Comment:
Readability?

```suggestion
        # If a change already exists for this change_id, user, and object, update it instead of creating a new one.
        # If the object was deleted then recreated with the same pk (don't do this), change the action to update.
```